packages/plugin-ext/src/plugin/tree/tree-views.ts

```diff
@@ -449,17 +449,17 @@ class TreeViewExtImpl<T> implements Disposable {
     }
 
     setSelection(selectedItemIds: string[]): void {
-        const toDelete = new Set<string>(this.selectedItemIds);
-        for (const id of this.selectedItemIds) {
-            toDelete.delete(id);
-            if (!this.selectedItemIds.has(id)) {
+        if (selectedItemIds.length !== this.selectedItemIds.size) {
```

Comment:
This is a somewhat counterintuitive check. I don't see any functional regression, so it must be fine, but just on the face of it this looks like it should do nothing as long as the size of the selection isn't changed - so if you're only selecting one item at a time, it looks like it should always be a no-op. What's going on in the background to make that naive impression incorrect?

Reply:
It looks like it's probably this phenomenon: https://github.com/eclipse-theia/theia/issues/9325#issuecomment-817710825 . But if `.size` is always going to be 0 - and if that's the reason this check works - then it seems better to be explicit and just write `0`.